src/ring_theory/witt_vector/perfect_ring.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2022 Robert Y. Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robert Y. Lewis, Heather Macbeth
+-/
+
+import ring_theory.witt_vector.domain
+import ring_theory.witt_vector.mul_coeff
+import tactic.linear_combination
+
+/-!
+
+# Witt vectors over a perfect ring
```

Comment:
I think this file should maybe be called `discrete_valuation_ring.lean`. And it should also be the main result of this file. You are really close, I'm sure we can knock off a proof.

Reply:
I pushed a proof

Reply:
Thanks Johan!!